File: mobile_app/test_ws.dart
import 'dart:io';

void main() async {
  try {
    final wsUrl = Uri.parse('ws://172.20.10.5:8002/ws/realtime/test');
    print('Connecting to $wsUrl');
    final socket = await WebSocket.connect(wsUrl.toString());
    print('Connected!');
    socket.listen(
      (data) => print('Data: $data'),
      onError: (err) => print('Error: $err'),
      onDone: () => print('Done: ${socket.closeCode} ${socket.closeReason}'),
    );
  } catch (e) {
    print('Exception: $e');
  }
}

File: backend/test_ws.py
import asyncio
import websockets

async def main():
    try:
        async with websockets.connect('ws://127.0.0.1:8002/ws/realtime/test') as ws:
            print("Connected!")
            await ws.send("hello")
    except Exception as e:
        print(f"Failed: {e}")

asyncio.run(main())
